test/functional/tests/initialize/test_startup_init_config.py

```diff
@@ -570,3 +573,127 @@ def validate_cache(cache_mode):
             f"Cache started in wrong mode!\n"
             f"Should start in {cache_mode}, but started in {current_mode} mode."
         )
+
+
+@pytest.mark.remote_only
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.parametrizex("reboot_type", ["soft", "hard"])
+def test_cas_lazy_startup_core_path_by_id(cache_mode, reboot_type):
+    """
+    title: Test for CAS startup when cores are set in config with wrong by-id path.
+    description: |
+      Start cache, add to config different links to devices that make up the cache
+      and check if cache start fails after reboot. Clear cache metadata before reboot.
+    pass_criteria:
+      - System does not crash
+      - Cache is running after startup
+      - Cores are detached after startup
+    """
+
+    with TestRun.step("Prepare partitions for cache and for cores"):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(200, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(400, Unit.MebiByte)] * cores_number)
+
+    with TestRun.step("Start cache and add cores"):
+        cores = []
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+        for i in range(cores_number):
+            cores.append(cache.add_core(core_dev.partitions[i]))
```

Comment:
You might use
```suggestion
        cache = casadm.start_cache(cache_part, cache_mode, force=True)
        cores = [cache.add_core(partition) for partition in core_dev.partitions]
```

Reply:
I'll change it like below, because I realized that 'cores' is not used anywhere.